src/mapping.ts

```diff
@@ -2,18 +2,33 @@ import { Address } from "@graphprotocol/graph-ts"
 import { TransferSingle, TransferBatch, URI, ERC1155 } from "../generated/Rarible/ERC1155";
 import { ERC721, Transfer } from "../generated/templates/NftContract/ERC721"
 import { NftContract, Nft } from "../generated/schema"
+import { ZERO_ADDRESS } from "./constants";
 
 export function handleTransfer(event: Transfer): void {
-  let id = event.address.toHexString() + "/" + event.params.id.toString();
+  let address = event.address.toHexString();
+
+  if (NftContract.load(address) == null) {
+    let nftContract = new NftContract(address);
+    nftContract.name = fetchName(event.address);
+    nftContract.symbol = fetchSymbol(event.address);
+    nftContract.platform = nftContract.name;
```

Comment:
For Nifty Gateway, we have names like "Dimenticati by Simone Vezzani"
So I'm not sure that it's gonna work as you intended, though it definitely looks better

Reply:
I reverted to my previous commit with the` handleTransferSuperRare`. Please check